### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by hassanShakur**
In this branch, I implemented the following requirements:

-  Create a method called `current_user` in your ApplicationController to make current user data available to all controllers.
    -  It will return the first user from the database.
    -  Don't worry about proper authentication at this time. You will add it in the upcoming projects.
- Create forms to perform the following functions:
    - Creates a Post on behalf of the `current_user` (use the method that you created in your ApplicationController).
    - Create a comment on behalf of the `current_user` (use the method that you created in your ApplicationController).
- Allow Users to add likes to Posts.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,10 +1,14 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
-  # Defines the root path route ("/")
+  resources :users, only: [:index, :show] do
+    resources :posts, only: [:index, :show, :new, :create] do
```
Comment (Shedrack-Sunday) [inline config/routes.rb:5]: -[ ] From your Routes, i can't find the `posts#new` and `posts#create` paths. The create should have a `POST` method

### Your view
image

File: config/routes.rb [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
+      resources :comments, only: [:new, :create]
```
Comment (Shedrack-Sunday) [inline config/routes.rb:6]: -  [ ] Just like for the post, you need endpoints for your comments. Though implemneted here, i can't find thein the routes log out output.

### Your view:
image

### Sample of expected:
```
   POST   /posts/:post_id/comments(.:format)     comments#create
 GET   /posts/:post_id/comments/new(.:format)     comments#new
```

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -6,4 +6,39 @@ def index
   def show
     @user = User.find(params[:id])
   end
+
+  def new
+    @user = User.new
+  end
+
```
Comment (Shedrack-Sunday) [inline app/controllers/users_controller.rb:13]: - [ ] After adding two users and running your app, it turns out their not been added to the database as `ids` were not even assigned to them. Kindly review this and make sure added users are been added

### View:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @hassanShakur         👋 

# CHANGES REQUIRED ♻️

Impressive work on the project so far👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with the Rails   features ✔️ 
- Nice use of Controller  parameters✔️ 

# Required Changes  🌵  

- [x] You've done well, However, kindly follow the suggestion on the right query pattern properly so your impressive work will have a better implementation 👍 


_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: config/routes.rb (hunk 2 above)
Comment (hassanShakur) [reply config/routes.rb:6]: Hey @Shedrack-Sunday, could the issue be from your end? I have rechecked my routes and the post actions for posts, comments, and likes are present.

image

I would appreciate your feedback if I am doing something wrong.

**Review by Ibizugbe — APPROVED** [review]
Hi @hassanShakur 👋🏼👋🏼👋🏼, 

approved


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights✅✅✅

- [x] The routes have been properly updated👍🏼.
- [x] List of users are properly rendered👍🏼.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**